internal/service/redshift/logging_test.go

```diff
@@ -0,0 +1,298 @@
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"log_destination_type",
+				},
+			},
+		},
+	})
+}
+
+func testAccCheckLoggingDestroy(ctx context.Context) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := acctest.Provider.Meta().(*conns.AWSClient).RedshiftClient(ctx)
+
+		for _, rs := range s.RootModule().Resources {
+			if rs.Type != "aws_redshift_logging" {
+				continue
+			}
+
+			_, err := tfredshift.FindLoggingByID(ctx, conn, rs.Primary.ID)
+			if errs.IsA[*retry.NotFoundError](err) {
+				return nil
+			}
+			if err != nil {
+				return create.Error(names.Redshift, create.ErrActionCheckingDestroyed, tfredshift.ResNameLogging, rs.Primary.ID, err)
+			}
+
+			return create.Error(names.Redshift, create.ErrActionCheckingDestroyed, tfredshift.ResNameLogging, rs.Primary.ID, errors.New("not destroyed"))
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckLoggingExists(ctx context.Context, name string, log *redshift.DescribeLoggingStatusOutput) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return create.Error(names.Redshift, create.ErrActionCheckingExistence, tfredshift.ResNameLogging, name, errors.New("not found"))
+		}
+
+		if rs.Primary.ID == "" {
+			return create.Error(names.Redshift, create.ErrActionCheckingExistence, tfredshift.ResNameLogging, name, errors.New("not set"))
+		}
+
+		conn := acctest.Provider.Meta().(*conns.AWSClient).RedshiftClient(ctx)
+		out, err := tfredshift.FindLoggingByID(ctx, conn, rs.Primary.ID)
+		if err != nil {
+			return create.Error(names.Redshift, create.ErrActionCheckingExistence, tfredshift.ResNameLogging, rs.Primary.ID, err)
+		}
+
+		*log = *out
+
+		return nil
+	}
+}
+
+func testAccLoggingConfigBase(rName string) string {
+	return acctest.ConfigCompose(
+		// "InvalidVPCNetworkStateFault: The requested AZ us-west-2a is not a valid AZ."
+		acctest.ConfigAvailableAZsNoOptInExclude("usw2-az2"),
+		fmt.Sprintf(`
+resource "aws_redshift_cluster" "test" {
+  cluster_identifier                  = %[1]q
+  availability_zone                   = data.aws_availability_zones.available.names[0]
+  database_name                       = "mydb"
+  master_username                     = "foo_test"
+  master_password                     = "Mustbe8characters"
+  multi_az                            = false
+  node_type                           = "dc2.large"
+  automated_snapshot_retention_period = 0
+  allow_version_upgrade               = false
+  skip_final_snapshot                 = true
+}
+`, rName))
+}
+
+func testAccLoggingConfig_basic(rName string) string {
+	return acctest.ConfigCompose(
+		testAccLoggingConfigBase(rName),
+		`
+resource "aws_redshift_logging" "test" {
+  cluster_identifier   = aws_redshift_cluster.test.id
+  log_destination_type = "cloudwatch"
+  log_exports          = ["connectionlog", "useractivitylog", "userlog"]
```

Comment:
A `basic` test shouldn't set `Optional` parameters, so that it tests defaults

Reply:
While technically optional, this argument is required when the `log_destination_type` is `cloudwatch`.